app/features/connect_with_them/tools.py

```diff
@@ -0,0 +1,137 @@
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import RequestException
+from langchain import hub
+from langchain.agents import AgentExecutor,create_react_agent,tool
+from langchain_community.document_loaders import WebBaseLoader
+from langchain.chains.summarize.chain import load_summarize_chain
+from langchain.tools.base import StructuredTool
+from langchain_groq import ChatGroq
+from langchain.pydantic_v1 import BaseModel, Field
+
+from app.features.connect_with_them.Credentials import credentials,categories_mapping
+from app.features.connect_with_them.prompt.Prompts import Prompt_query
+
+from bs4 import BeautifulSoup
+
+model_name = 'llama-3.1-70b-versatile'
+
+llm = ChatGroq(model=model_name,temperature=0.3,api_key="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")
```

Comment:
Remove the api_key and use the .env file values. It's dangerous to hardcode your api key.